performance-tests/bench/tcp_latency/tcp_latency.cpp

```diff
@@ -0,0 +1,251 @@
+
+  return 0;
+}
+
+static void
+receiver(ACE_SOCK_Stream& cli_stream)
+{
+  ACE_INET_Addr peer_addr;
+  DataSample data;
+  ssize_t rcv_cnt = 1;
+
+  Bench::WorkerReport report{};
+  report.process_report.participants.length(1);
+  report.process_report.participants[0].subscribers.length(1);
+  report.process_report.participants[0].subscribers[0].datareaders.length(1);
+  Builder::PropertySeq& properties = report.process_report.participants[0].subscribers[0].datareaders[0].properties;
+
+  const size_t buffer_size = static_cast<size_t>(1.1 * frequency * test_duration);
+
+  auto round_trip_latency_stat_block = std::make_shared<Bench::PropertyStatBlock>(properties, "round_trip_latency", buffer_size);
+
+  while (rcv_cnt) {
+    rcv_cnt = cli_stream.recv_n (&data,
+                                 message_size);
+    if (rcv_cnt > 0) {
+      round_trip_latency_stat_block->update(5E-10 * (ACE_OS::gethrtime() - data.sending_time_));
+    }
+    else if (rcv_cnt < 0) {
+      ACE_ERROR((LM_ERROR,
+                 ACE_TEXT("(%P|%t) %p\n"),
+                 ACE_TEXT("cli_stream.recv_n")));
+      return;
+    }
+  }
+
+  round_trip_latency_stat_block->finalize();
+
+  if (!report_file_name.empty()) {
+    ofstream report_file(report_file_name);
+    if (report_file.good()) {
+      idl_2_json(report, report_file, 9u);
+    }
+  }
+
+  return;
+}
+
+
+int echo_server(int port)
+{
+
+  // Acceptor
+  ACE_SOCK_Acceptor peer_acceptor;
+
+  ACE_INET_Addr server_addr, cli_addr;
+  server_addr.set (port);
+  ACE_SOCK_Stream server_stream;
+
+  if (peer_acceptor.open (server_addr) == -1)
+  {
+    ACE_ERROR((LM_ERROR,
+               ACE_TEXT("(%P|%t) %p\n"),
+               ACE_TEXT("peer_acceptor.open")));
+    return 1;
+  }
+
+  if (peer_acceptor.accept (server_stream, &cli_addr) == -1)
+  {
+    ACE_ERROR((LM_ERROR,
+               ACE_TEXT("(%P|%t) %p\n"),
+               ACE_TEXT("peer_acceptor.accept")));
+    return 1;
+  }
+
+  DataSample data;
+  ACE_Time_Value timeout(120,0);
+  ssize_t rcv_cnt = 1;
+
+  while (rcv_cnt) {
+    rcv_cnt = server_stream.recv_n (&data,
+                                    message_size);
+    if (rcv_cnt > 0) {
+      if (server_stream.send_n(&data,
+                               message_size) < 0){
+        ACE_ERROR((LM_ERROR,
+                   ACE_TEXT("(%P|%t) %p\n"),
+                   ACE_TEXT("server_stream.send_n")));
+        return 1;
+      }
+    }
+    else if (rcv_cnt < 0) {
+      ACE_ERROR((LM_ERROR,
+                 ACE_TEXT("(%P|%t) %p\n"),
+                 ACE_TEXT("server_stream.recv_n")));
+      return 1;
+    }
+    else {
+      server_stream.close_writer();
+    }
+  }
+  return 0;
+}
+
+int parse_args(int argc, ACE_TCHAR ** argv)
+{
+  int c;
+
+  ACE_Get_Opt getopt(argc, argv, "c:d:f:m:s:r:");
+  bool ok = true;
+
+  while (ok && (c = getopt()) != -1) {
+    switch (c) {
+      case 'c':
+        ok = (server_addr.set(getopt.opt_arg()) == 0);
+        break;
```

Comment:
```suggestion
  while (ok && (c = getopt()) != -1) {
    switch (c) {
    case 'c':
      ok = (server_addr.set(getopt.opt_arg()) == 0);
      break;
```

and for each other case

Reply:
Personal preference would be to leave the indentation for clarity. I don't think the guidelines has anything for this?

Reply:
It doesn't.  I checked because I know in the past I've been told to match case to switch but some places in the code have it the way you do.

Reply:
I'm fine with leaving it as is because there isn't risk of more depth being added in the future, which I think is the reason for putting the cases next to the switch.